dmrpp_generator/dmrpp.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import re
+
+import boto3
+import requests
+
+
+class DMRpp:
+    def __init__(self) -> None:
+        self.s3_client = boto3.client('s3')
+        self.session = requests.Session()
+
+    def get_https_file(self, url, host_path='/tmp'):
+        self.get_http_file(url, host_path)
+
+    def get_http_file(self, url, host_path="/tmp"):
+        """
+        Downloads a file from the provided url.
+        :param url: Location to find the file to download http or https
+        :param host_path: Where to store the downloaded file
+        """
+        filename = url.rsplit('/', 1)[-1]
+        try:
+            response = self.session.get(url)
+            with open(f'{host_path}/{filename}', 'wb') as file:
+                file.write(response.content)
+        except Exception as e:
+            logging.log(level=20, msg=str(e))
+
+    def get_s3_file(self, s3_link, host_path="/tmp"):
+        """
+        Downloads a file from the provided url.
+        :param s3_link: Location to find the file to download from S3
+        :param host_path: Where to store the downloaded file
+        """
+        filename = s3_link.rsplit('/', 1)[-1]
+        reg_res = re.match(rf'^.*://([^/]*)/(.*)', s3_link)
+        bucket_name = reg_res.group(1)
+        key = reg_res.group(2)
+        try:
+            self.s3_client.download_file(bucket_name, key, f'{host_path}/{filename}')
```

Comment:
Can we check here if the file already exist to not download it? same thing for HTTP protocol